fastdds_python/src/swig/fastdds/rtps/common/InstanceHandle.i

```diff
@@ -26,60 +29,119 @@ long hash(const eprosima::fastdds::rtps::InstanceHandle_t& handle)
-                    buf, buf+1, buf+2, buf+3, buf+4, buf+5, buf+6, buf+7, buf+8,
-                    buf+9, buf+10, buf+11, buf+12, buf+13, buf+14, buf+15))
+%ignore eprosima::fastdds::rtps::InstanceHandleValue_t::operator const octet* () const;
+%ignore eprosima::fastdds::rtps::InstanceHandleValue_t::operator octet* ();
+
+%extend eprosima::fastdds::rtps::InstanceHandleValue_t {
+
+    // Constructor from sequence of 16 bytes (tupla/lista/bytes/bytearray)
+    InstanceHandleValue_t(PyObject* seq) {
+        eprosima::fastdds::rtps::InstanceHandleValue_t* self = new eprosima::fastdds::rtps::InstanceHandleValue_t();
+        SWIG_PYTHON_THREAD_BEGIN_BLOCK;
+
+        // Fast-path: bytes
+        if (PyBytes_Check(seq)) {
+            if (PyBytes_GET_SIZE(seq) == 16)
+            {
+                const char* b = PyBytes_AS_STRING(seq);
+                for (int i = 0; i < 16; ++i) (*self)[i] = (uint8_t)(unsigned char)b[i];
+            }
+            else
+            {
+                delete self;
+                self = nullptr;
+                PyErr_SetString(PyExc_ValueError, "Expected 16 bytes");
+            }
+        }
+        // Fast-path: bytearray
+        else if (PyByteArray_Check(seq))
         {
-            PyErr_SetString(PyExc_TypeError, "tuple must have 16 elements");
-            SWIG_fail;
+            if (PyByteArray_GET_SIZE(seq) == 16)
+            {
+                const char* b = PyByteArray_AS_STRING(seq);
+                for (int i = 0; i < 16; ++i) (*self)[i] = (uint8_t)(unsigned char)b[i];
+            }
+            else
+            {
+                delete self;
+                self = nullptr;
+                PyErr_SetString(PyExc_ValueError, "Expected 16 bytes");
+            }
         }
-        $1 = &temp;
-    }
-    else
-    {
-        PyErr_SetString(PyExc_TypeError, "expected a tuple.");
-        SWIG_fail;
+        else
+        {
+            // Generic fallback: iterable from 16 ints 0..255
+            PyObject* it = PyObject_GetIter(seq);
+            size_t count {0};
+            if (it)
+            {
+                PyObject* item {nullptr};
+                while ((item = PyIter_Next(it)))
+                {
+                    long val = PyLong_AsLong(item);
+                    Py_DECREF(item);
+                    if (val == -1 && PyErr_Occurred())
+                    {
+                        delete self;
+                        self = nullptr;
+                        PyErr_SetString(PyExc_TypeError, "Sequence must contain integers");
+                        break;
+                    }
+                    else if (val < 0 || val > 255)
+                    {
+                        delete self;
+                        self = nullptr;
+                        PyErr_SetString(PyExc_ValueError, "Each value must be in 0..255");
+                        break;
+                    }
+
+                    (*self)[count] = static_cast<uint8_t>(val);
```

Comment:
This might produce a wrong access if `count >= 16`

Reply:
Done in 57a4f12